### DISCUSSION TITLE
Major changes after v0.4.1. Is the next release close?

### DISCUSSION THREAD

**Discussion by azzamsa** [KDAB/cxx-qt#432] [Q&A] [answered]
Hi.

I will write a blog post on how to build a Qt app with Rust using cxx-qt. It is the only actively maintained crate so far.
I see the API in the `main` branch is simpler. 

- The `run.cpp` changed to `register_types.cpp`
  - https://github.com/KDAB/cxx-qt/blob/v0.4.1/examples/cargo_without_cmake/cpp/run.cpp
  -  https://github.com/KDAB/cxx-qt/blob/d1c4b5e926c896690c129f5e1b29dfdf5b282984/examples/cargo_without_cmake/cpp/register_types.cpp#L1 
  
- A lot of API changes in `CxxQtBuilder`
  - https://github.com/KDAB/cxx-qt/blob/d1c4b5e926c896690c129f5e1b29dfdf5b282984/examples/cargo_without_cmake/build.rs#L1
  
Is the next major release close? Something like v0.5.0?

Thanks for maintaining cxx-qt :heart:

**Comment by Be-ing** [comment, marked as answer]
`type_registration.cpp` is gone in https://github.com/KDAB/cxx-qt/pull/428 :grin: 

> Is the next major release close? Something like v0.5.0?

Yes. Hopefully in the next week or two. I think we'll be in a good place for a release after merging the currently open pull requests.

**Reply by LeonMatthesKDAB** [reply]
Completely agree with @Be-ing - in addition to the great reduction in necessary C++ code we'll have examples of a full QAbstractItemModel subclass in 0.5.0 as well. (see #363 ).

**Reply by azzamsa** [reply]
> great reduction in necessary C++ code

This is the main reason I wait for a new release :grin:
